src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/ProvidePublicParameterlessSafeHandleConstructor.Fixer.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices
+{
+    public abstract class ProvidePublicParameterlessSafeHandleConstructorFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(ProvidePublicParameterlessSafeHandleConstructorAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/main/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxGenerator generator = SyntaxGenerator.GetGenerator(context.Document);
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            SyntaxNode enclosingNode = root.FindNode(context.Span);
+            SyntaxNode declaration = generator.GetDeclaration(enclosingNode);
+            if (declaration == null)
+            {
+                return;
+            }
+
+            SemanticModel model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+            ISymbol symbol = model.GetDeclaredSymbol(declaration, context.CancellationToken);
+            if (symbol is IMethodSymbol method)
+            {
+                Diagnostic diagnostic = context.Diagnostics.First();
+                string title = MicrosoftNetCoreAnalyzersResources.ProvidePublicParameterlessSafeHandleConstructorTitle;
+                context.RegisterCodeFix(
+                    new MyCodeAction(
+                        title,
+                        async ct => await MakeParameterlessConstructorPublic(declaration, context.Document, context.CancellationToken).ConfigureAwait(false),
+                        equivalenceKey: title),
```

Comment:
Both forms are used in the repository, but this one should be the recommended form.
Making equivalence keys non-localized can help with future features from roslyn side.

```suggestion
                        equivalenceKey: nameof(MicrosoftNetCoreAnalyzersResources.ProvidePublicParameterlessSafeHandleConstructorTitle)),
```